app/Console/Commands/GeneratePdfs.php

```diff
@@ -105,20 +104,38 @@ protected function generatePdf($model, $route = null)
         }
 
         set_time_limit(0);
-        $html = file_get_contents($fullUrl . '?print=true');
+        $html = Http::get($fullUrl, ['print' => true])->body();
 
-        $pdfFileName = 'download-' . $route . '-' . $model->id . '.pdf';
+        $pdfFileName = self::pdfFileName($model);
         $pdfPath = storage_path('app/' . $pdfFileName);
         $prince->convertStringToFile($html, $pdfPath);
 
         if (config('aic.pdf_s3_enabled')) {
             // Stream the file to S3; be sure to set `AWS_BUCKET` in `.env` and otherwise configure credentials
-            Storage::disk('pdf_s3')->putFileAs('/pdf/static', new File($pdfPath), $pdfFileName, 'public');
+            Storage::disk('pdf_s3')->putFileAs(self::STORAGE_PATH, new File($pdfPath), $pdfFileName, 'public');
         }
 
-        if ($model->pdf_download_path != '/pdf/static/' . $pdfFileName) {
-            $model->pdf_download_path = '/pdf/static/' . $pdfFileName;
+        if ($model->pdf_download_path != self::pdfDownloadPath($pdfFileName)) {
+            $model->pdf_download_path = self::pdfDownloadPath($pdfFileName);
             $model->save();
         }
+        $class = class_basename($model);
+        $this->info("Generated PDF for {$class} with ID {$model->id}");
+    }
+
+    public static function pdfFileName(Model $model): string
```

Comment:
Would this work using `App\Models\AbstractModel` instead of the Twill's model? To namespace this code to our app's model's specifically.

Reply:
Makes sense, updated